client/src/utils/validators.js

```diff
@@ -10,11 +11,24 @@ const validateTopLevelDomain = (string) => {
   return false;
 };
 
+const validateDisposableEmail = (email) => {
+  for (const wildcard of wildcards) {
```

Comment:
Let's change it to this for a O(1) lookup
```suggestion
const emailDomain = email.split("@")[1];
const wildcardsSet = new Set(wildcards);
return wildcardsSet.has(emailDomain);

```

Reply:
Thank you for indicating this, so making a new Set() is also O(n), too, you are right about [Set.prototype.has()](https://adrianmejia.com/data-structures-time-complexity-for-beginners-arrays-hashmaps-linked-lists-stacks-queues-tutorial/#Set-Operations-runtime) is O(1) (not worst case) but when I check the performance of both code I found out that current for loop with an includes is faster, can you check that too? 

```
const validateDisposableEmail = (email) => {
  console.time("LOOP");
  for (const wildcard of wildcards) {
    if (email.includes(wildcard)) {
      console.log("includes");
    }
  }
  console.timeEnd("LOOP");
};
```
![2021-01-03_15-30-03](https://user-images.githubusercontent.com/34690158/103478593-c519a900-4dd8-11eb-904f-ffd42d840316.gif)

```
 const validateDisposableEmail = (email) => {
  console.time("SET");
  const emailDomain = email.split("@")[1];
  const wildcardsSet = new Set(wildcards);
  console.log(wildcardsSet.has(emailDomain));
  console.timeEnd("SET");
};
```
![2021-01-03_15-34-02](https://user-images.githubusercontent.com/34690158/103478648-35282f00-4dd9-11eb-8898-346c824af3cf.gif)

Reply:
I guess @BMariscal meant to build the Set outside of the `validateDisposableEmail()` function (maybe pass the set as the input)

Reply:
> I guess @BMariscal meant to build the Set outside of the `validateDisposableEmail()` function

make sense, I don't know why I stuck to that function to build the Set. Anyway, can you review, please?
@BMariscal @ddveloper